pennylane/qaoa/cost.py

```diff
@@ -56,26 +208,15 @@ def maxcut(graph):
     >>> cost_h, mixer_h = qml.qaoa.maxcut(graph)
     >>> print(cost_h)
     >>> print(mixer_h)
-    (-0.5) [I0 I1] + (0.5) [Z0 Z1] + (-0.5) [I1 I2] + (0.5) [Z1 Z2]
+    (0.5) [Z0 Z1] + (0.5) [Z1 Z2] + (-1.0) [I0]
     (1.0) [X0] + (1.0) [X1] + (1.0) [X2]
     """
 
     if not isinstance(graph, nx.Graph):
-        raise ValueError(
-            "Input graph must be a nx.Graph object, got {}".format(type(graph).__name__)
-        )
-
-    edges = graph.edges
-
-    coeffs = []
-    obs = []
-
-    for node1, node2 in edges:
-
-        obs.append(qml.Identity(node1) @ qml.Identity(node2))
-        coeffs.append(-0.5)
-
-        obs.append(qml.PauliZ(node1) @ qml.PauliZ(node2))
-        coeffs.append(0.5)
+        raise ValueError("Input graph must be a nx.Graph, got {}".format(type(graph).__name__))
 
-    return (qml.Hamiltonian(coeffs, obs), qaoa.x_mixer(graph.nodes))
+    identity_h = qml.Hamiltonian(
+        [-0.5 for e in graph.edges], [qml.Identity(e[0]) @ qml.Identity(e[1]) for e in graph.edges]
+    )
+    H = edge_driver(graph, ["10", "01"]) + identity_h
+    return (H, qaoa.x_mixer(graph.nodes))
```

Comment:
Was the approach before incorrect? It looks like more than just changing the method, but the actual result is different?

Reply:
oh maybe just the identities were different :thinking:

Reply:
@trbromley Yes, the only difference is the identities, which get added together (as the new approach to constructing this Hamiltonian involves addition of terms, which automatically simplifies the newly-created Hamiltonian)